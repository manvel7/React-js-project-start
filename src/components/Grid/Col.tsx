import {HTMLAttributes} from "react"
import clsx from "clsx"
import _ from "lodash"
import styled, {css, StyledProps} from "styled-components"
import {ColBreakpointName} from "../../interfaces/grid"
import {mediaBreakpointUp} from "../../utils/breakpoint"
import {pxToRem, toPercent} from "../../utils/unit"

const colBreakpoints: Array<ColBreakpointName> = ["xs", "sm", "md", "lg", "xl"]

type ColSize = number | "auto" | true
type ColOrder = number | "first" | "last"

interface IColConfig {
  size?: ColSize
  order?: ColOrder
  offset?: number
}

export interface IProps extends HTMLAttributes<HTMLDivElement> {
  xs?: IColConfig
  sm?: IColConfig
  md?: IColConfig
  lg?: IColConfig
  xl?: IColConfig
}

const widthMixin = css<StyledProps<IProps>>`
  ${(props) => {
    const {
      theme: {
        grid: {breakpoints, columns}
      }
    } = props

    return colBreakpoints
      .map((breakpoint) => {
        const colConfig = props[breakpoint]

        if (!_.isNil(colConfig)) {
          const {size, order, offset} = colConfig
          const breakpointStyles: string[] = []

          // Column size
          switch (true) {
            case size === "auto":
              breakpointStyles.push(
                mediaBreakpointUp(
                  breakpoint,
                  breakpoints,
                  `flex: 0 0 auto;
                  width: auto;
                  max-width: none;`
                )
              )
              break
            case _.isNumber(size):
              breakpointStyles.push(
                mediaBreakpointUp(
                  breakpoint,
                  breakpoints,
                  `flex: 0 0 ${toPercent(size as number, columns)};
                  max-width: ${toPercent(size as number, columns)};`
                )
              )
              break
            default:
              breakpointStyles.push(
                mediaBreakpointUp(
                  breakpoint,
                  breakpoints,
                  `flex-basis: 0;
                  flex-grow: 1;
                  max-width: 100%;`
                )
              )
          }

          // Column order
          switch (true) {
            case order === "first":
              breakpointStyles.push(
                mediaBreakpointUp(breakpoint, breakpoints, "order: -1;")
              )
              break
            case order === "last":
              breakpointStyles.push(
                mediaBreakpointUp(breakpoint, breakpoints, "order: 13;")
              )
              break
            case _.isNumber(order):
              breakpointStyles.push(
                mediaBreakpointUp(breakpoint, breakpoints, `order: ${order};`)
              )
              break
            default:
          }

          // Column offset
          if (!_.isNil(offset) && _.isNumber(offset)) {
            breakpointStyles.push(
              mediaBreakpointUp(
                breakpoint,
                breakpoints,
                `margin-left: ${toPercent(offset, columns)};`
              )
            )
          }

          return breakpointStyles.join("\n")
        }

        return mediaBreakpointUp(
          breakpoint,
          breakpoints,
          `flex-basis: 0;
          flex-grow: 1;
          max-width: 100%;`
        )
      })
      .join("\n")
  }}
`

const Col = styled.div.attrs<IProps, IProps>((props) => {
  const {className, xs, sm, md, lg, xl} = props

  return {
    className: clsx(className, {
      col: true,
      // xs
      [`col-${xs?.size}`]: !_.isNil(xs?.size) && xs?.size !== true,
      [`order-xs-${xs?.order}`]: !_.isNil(xs?.order),
      [`offset-xs-${xs?.offset}`]: !_.isNil(xs?.offset),
      // sm
      "col-sm": !_.isNil(sm?.size) && sm?.size === true,
      [`col-sm-${sm?.size}`]: !_.isNil(sm?.size) && sm?.size !== true,
      [`order-sm-${sm?.order}`]: !_.isNil(sm?.order),
      [`offset-sm-${sm?.offset}`]: !_.isNil(sm?.offset),
      // md
      "col-md": !_.isNil(md?.size) && md?.size === true,
      [`col-md-${md?.size}`]: !_.isNil(md?.size) && md?.size !== true,
      [`order-md-${md?.order}`]: !_.isNil(md?.order),
      [`offset-md-${md?.offset}`]: !_.isNil(md?.offset),
      // lg
      "col-lg": !_.isNil(lg?.size) && lg?.size === true,
      [`col-lg-${lg?.size}`]: !_.isNil(lg?.size) && lg?.size !== true,
      [`order-lg-${lg?.order}`]: !_.isNil(lg?.order),
      [`offset-lg-${lg?.offset}`]: !_.isNil(lg?.offset),
      // xl
      "col-xl": !_.isNil(xl?.size) && xl?.size === true,
      [`col-xl-${xl?.size}`]: !_.isNil(xl?.size) && xl?.size !== true,
      [`order-xl-${xl?.order}`]: !_.isNil(xl?.order),
      [`offset-xl-${xl?.offset}`]: !_.isNil(xl?.offset)
    })
  }
})`
  min-height: 0.1rem;
  padding-left: calc(
    ${({
        theme: {
          grid: {gutterWidth},
          typography: {htmlFontSize}
        }
      }) => pxToRem(gutterWidth, htmlFontSize)} / 2
  );
  padding-right: calc(
    ${({
        theme: {
          grid: {gutterWidth},
          typography: {htmlFontSize}
        }
      }) => pxToRem(gutterWidth, htmlFontSize)} / 2
  );
  position: relative;
  width: 100%;
  ${widthMixin}
`

export default Col
