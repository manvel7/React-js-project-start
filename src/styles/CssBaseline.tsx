import {
  createGlobalStyle,
  css,
  DefaultTheme,
  GlobalStyleComponent,
  StyledProps
} from "styled-components"
import {mediaBreakpointDown} from "../utils/breakpoint"
import {getValue, toPercent} from "../utils/unit"
import {
  bodyMixin,
  smallMixin,
  h1Mixin,
  h2Mixin,
  h3Mixin,
  h4Mixin,
  h5Mixin,
  h6Mixin
} from "./typography"

interface IProps {}

const genericResetMixin = css<StyledProps<IProps>>`
  html,
  body,
  div,
  span,
  applet,
  object,
  iframe,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  p,
  blockquote,
  pre,
  a,
  abbr,
  acronym,
  address,
  big,
  cite,
  code,
  del,
  dfn,
  em,
  img,
  ins,
  kbd,
  q,
  s,
  samp,
  small,
  strike,
  strong,
  sub,
  sup,
  tt,
  var,
  b,
  u,
  i,
  center,
  dl,
  dt,
  dd,
  ol,
  ul,
  li,
  fieldset,
  form,
  label,
  legend,
  table,
  caption,
  tbody,
  tfoot,
  thead,
  tr,
  th,
  td,
  article,
  aside,
  canvas,
  details,
  embed,
  figure,
  figcaption,
  footer,
  header,
  hgroup,
  menu,
  nav,
  output,
  ruby,
  section,
  summary,
  time,
  mark,
  audio,
  video {
    margin: 0;
    padding: 0;
    border: 0;
    vertical-align: baseline;
  }

  article,
  aside,
  details,
  figcaption,
  figure,
  footer,
  header,
  hgroup,
  menu,
  nav,
  section {
    display: block;
  }

  ol,
  ul {
    list-style: none;
  }

  blockquote,
  q {
    quotes: none;
  }

  blockquote:before,
  blockquote:after,
  q:before,
  q:after {
    content: "";
    content: none;
  }

  table {
    border-collapse: collapse;
    border-spacing: 0;
  }
`

const appResetMixin = css<StyledProps<IProps>>`
  html {
    /* stylelint-disable property-no-vendor-prefix */
    -ms-overflow-style: scrollbar; /* Set box-sizing so that the width is not affected by padding or border */
    -ms-text-size-adjust: 100%; /* Set default font family in all browsers. */
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0); /* Makes 1rem = 10px */
    -webkit-text-size-adjust: 100%; /* Prevent adjustments of font size after orientation changes in IE on Windows Phone and in iOS. */
    /* stylelint-enable property-no-vendor-prefix */
    box-sizing: border-box;
  }

  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }

  body {
    background-color: ${({theme}) => theme.palette.background.white};

    /* Hide overflow content */
    &.overflow {
      overflow: hidden;
    }

    /* Prevent focus outline on elements that can't be accessed via keyboard. */
    [tabindex="-1"]:focus {
      outline: none !important;
    }

    /* Hide overflow in IE */
    svg:not(:root) {
      overflow: hidden;
    }

    /* Hide elements has 'hidden' attribute in IE 10- */
    [hidden] {
      display: none !important;
    }

    ::-webkit-file-upload-button {
      /* stylelint-disable-next-line property-no-vendor-prefix */
      -webkit-appearance: button;
      font: inherit;
    }
  }
`

const typographyMixin = css<StyledProps<IProps>>`
  html {
    font-family: ${({theme}) =>
      theme.typography
        .fontFamily}; /* Prevent scrollbars to overlap content in IE11 and Edge */
    font-size: ${({theme}) =>
      toPercent(
        getValue(theme.typography.htmlFontSize),
        16
      )}; /* Set default tap highlight to be transparent in iOS. */
    font-style: normal;
    color: ${({theme}) => theme.palette.text.dark};
  }

  body {
    ${bodyMixin<IProps>()}
  }

  h1 {
    ${h1Mixin<IProps>()}
  }

  h2 {
    ${h2Mixin<IProps>()}
  }

  h3 {
    ${h3Mixin<IProps>()}
  }

  h4 {
    ${h4Mixin<IProps>()}
  }

  h5 {
    ${h5Mixin<IProps>()}
  }

  h6 {
    ${h6Mixin<IProps>()}
  }

  small {
    ${smallMixin<IProps>()}
  }

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.bold};

    h1,
    h2,
    h3,
    h4,
    h5,
    h6,
    caption {
      font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
    }
  }
`

const utilsMixin = css<StyledProps<IProps>>`
  .hidden {
    visibility: hidden;
  }

  .d-none {
    display: none;
  }

  .hidden-sm-down {
    ${({theme}) =>
      mediaBreakpointDown(
        "sm",
        theme.grid.breakpoints,
        `
      display: none;
    `
      )}
  }

  .text-center {
    text-align: center;
  }
`

const CssBaseline: GlobalStyleComponent<
  IProps,
  DefaultTheme
> = createGlobalStyle`
  ${genericResetMixin}

  ${appResetMixin}

  ${typographyMixin}

  ${utilsMixin}
`

export default CssBaseline
