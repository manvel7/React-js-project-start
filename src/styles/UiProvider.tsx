import React, {FunctionComponent} from "react"
import {ThemeProvider} from "styled-components"
import theme from "./theme"
import {CssBaseline} from "./index"

interface IProps {}

const UiProvider: FunctionComponent<IProps> = (props) => {
  const {children} = props

  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      {children}
    </ThemeProvider>
  )
}

export default UiProvider
