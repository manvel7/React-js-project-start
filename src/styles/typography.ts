import {
  css,
  DefaultTheme,
  FlattenInterpolation,
  StyledProps,
  ThemedStyledProps
} from "styled-components"
import {pxToRem} from "../utils/unit"

const h1Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h1.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h1.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h1.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h1.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h2Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h2.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h2.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h2.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h2.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h3Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h3.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h3.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h3.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h3.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h4Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h4.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h4.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h4.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h4.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h5Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h5.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h5.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h5.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h5.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h6Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h6.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h6.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h6.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h6.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const bodyMixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(
      theme.typography.variants.body.size,
      theme.typography.htmlFontSize
    )};
  font-weight: ${({theme}) => theme.typography.variants.body.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.body.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.body.lineHeight,
      theme.typography.htmlFontSize
    )};
`

const smallMixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(
      theme.typography.variants.caption.size,
      theme.typography.htmlFontSize
    )};
  font-weight: ${({theme}) => theme.typography.variants.caption.weight};
  letter-spacing: ${({theme}) =>
    theme.typography.variants.caption.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.caption.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`
export {
  h1Mixin,
  h2Mixin,
  h3Mixin,
  h4Mixin,
  h5Mixin,
  h6Mixin,
  bodyMixin,
  smallMixin
}
