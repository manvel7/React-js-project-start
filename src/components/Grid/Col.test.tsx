import React from "react"
import {render} from "@testing-library/react"
import {theme} from "../../styles"
import Col from "./Col"

describe("Col", () => {
  describe("with auto layout", () => {
    it("should match snapshot", () => {
      const result = render(<Col theme={theme}>This is a column</Col>)
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })

  describe("with size", () => {
    it("should match snapshot", () => {
      const result = render(
        <Col theme={theme} sm={{size: 3}}>
          This is a column
        </Col>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })

  describe("with offset", () => {
    it("should match snapshot", () => {
      const result = render(
        <Col theme={theme} sm={{offset: 3}}>
          This is a column
        </Col>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })

  describe("with order", () => {
    it("should match snapshot", () => {
      const result = render(
        <Col theme={theme} sm={{order: 3}}>
          This is a column
        </Col>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })

  describe("with size, offset and order", () => {
    it("should match snapshot", () => {
      const result = render(
        <Col theme={theme} sm={{size: 3, offset: 5, order: "last"}}>
          This is a column
        </Col>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })
})
