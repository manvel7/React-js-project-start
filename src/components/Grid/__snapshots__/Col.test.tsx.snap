// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col with auto layout should match snapshot 1`] = `
.c0 {
  min-height: 0.1rem;
  padding-left: calc( 3.2rem / 2 );
  padding-right: calc( 3.2rem / 2 );
  position: relative;
  width: 100%;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

@media (min-width:576px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

@media (min-width:992px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

@media (min-width:1200px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

<div
  class="c0 col"
>
  This is a column
</div>
`;

exports[`Col with offset should match snapshot 1`] = `
.c0 {
  min-height: 0.1rem;
  padding-left: calc( 3.2rem / 2 );
  padding-right: calc( 3.2rem / 2 );
  position: relative;
  width: 100%;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

@media (min-width:576px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

@media (min-width:576px) {
  .c0 {
    margin-left: 25%;
  }
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

@media (min-width:992px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

@media (min-width:1200px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

<div
  class="c0 col offset-sm-3"
>
  This is a column
</div>
`;

exports[`Col with order should match snapshot 1`] = `
.c0 {
  min-height: 0.1rem;
  padding-left: calc( 3.2rem / 2 );
  padding-right: calc( 3.2rem / 2 );
  position: relative;
  width: 100%;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

@media (min-width:576px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

@media (min-width:576px) {
  .c0 {
    -webkit-order: 3;
    -ms-flex-order: 3;
    order: 3;
  }
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

@media (min-width:992px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

@media (min-width:1200px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

<div
  class="c0 col order-sm-3"
>
  This is a column
</div>
`;

exports[`Col with size should match snapshot 1`] = `
.c0 {
  min-height: 0.1rem;
  padding-left: calc( 3.2rem / 2 );
  padding-right: calc( 3.2rem / 2 );
  position: relative;
  width: 100%;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

@media (min-width:576px) {
  .c0 {
    -webkit-flex: 0 0 25%;
    -ms-flex: 0 0 25%;
    flex: 0 0 25%;
    max-width: 25%;
  }
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

@media (min-width:992px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

@media (min-width:1200px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

<div
  class="c0 col col-sm-3"
>
  This is a column
</div>
`;

exports[`Col with size, offset and order should match snapshot 1`] = `
.c0 {
  min-height: 0.1rem;
  padding-left: calc( 3.2rem / 2 );
  padding-right: calc( 3.2rem / 2 );
  position: relative;
  width: 100%;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

@media (min-width:576px) {
  .c0 {
    -webkit-flex: 0 0 25%;
    -ms-flex: 0 0 25%;
    flex: 0 0 25%;
    max-width: 25%;
  }
}

@media (min-width:576px) {
  .c0 {
    -webkit-order: 13;
    -ms-flex-order: 13;
    order: 13;
  }
}

@media (min-width:576px) {
  .c0 {
    margin-left: 41.6666666666%;
  }
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

@media (min-width:992px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

@media (min-width:1200px) {
  .c0 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

<div
  class="c0 col col-sm-3 order-sm-last offset-sm-5"
>
  This is a column
</div>
`;
